internal/rbd/controllerserver.go

```diff
@@ -318,12 +316,38 @@ func (cs *ControllerServer) CreateVolume(
 	return buildCreateVolumeResponse(req, rbdVol), nil
 }
 
-func flattenParentImage(ctx context.Context, rbdVol *rbdVolume, cr *util.Credentials) error {
+// flattenParentImage is to be called before proceeding with creating volume,
+// with datasource. This function flattens the parent image accordingly to
+// make sure no flattening is required during or after the new volume creation.
+// For parent volume, it's parent(temp clone or snapshot) is flattened.
+// For parent snapshot, the snapshot itself is flattened.
+func flattenParentImage(
+	ctx context.Context,
+	rbdVol *rbdVolume,
+	rbdSnap *rbdSnapshot,
+	cr *util.Credentials) error {
+	// flatten the image's parent before the reservation to avoid
+	// stale entries in post creation if we return ABORT error and the
+	// delete volume is not called
+	// reducing the limit for cloned images to make sure the limit is in range,
```

Comment:
@Rakshith-R nit: can you allign the lines 331 and 332?

Reply:
added a `.` its a different para